src/app/bookshelf/stats/BookshelfStats.tsx

```diff
@@ -0,0 +1,53 @@
+import { getTranslations } from "next-intl/server";
+import { getBookshelfStats } from "./getBookshelfStats";
+import StatsOverview from "./StatsOverview";
+import Header from "@/components/headers/Header";
+import Charts from "./Charts";
+import styles from "./BookshelfStats.module.css";
+
+export default async function BookshelfStats() {
+  const t = await getTranslations();
+
+  const books = await getBookshelfStats();
+
+  if (!books) {
+    return (
+      <div className={styles.statsContainer}>
```

Comment:
This looks like a piece of code that could be moved to the main `return` below, provided that you use optional access (or default value) for `books` - when using functional components, it's usually preferable to have just one return, due to how easy it is to forget about hook rules ;)